### ISSUE TITLE
[2.1] Change services to stop using "use" in as many places as possible

### ISSUE THREAD

**Issue by dschissler** [phalcon/phalcon-devtools#682]
So this would change from:

```php
$di->setShared('db', function () use ($config) {
```

to

```php
$di->setShared('db', function () {
    $config = $this->getConfig();
```

Would a pull request be welcome?

**Comment by sergeyklay** [comment]
I like it :)

**Comment by dschissler** [comment]
What about the config service?  Should it read from `'./'` in the service definition?  I do that in my project and I don't think that it is much of a stretch that the config file will be in the same place as the service definition.

**Comment by sergeyklay** [comment]
```php
use Phalcon\Config;
use RuntimeException;

/**
 * Prepare and return the Config.
 *
 * @param  string $path Config path [Optional]
 * @return Config
 *
 * @throws RuntimeException
 */
protected function initConfig($path = null)
{
    $path = $path ?: DOCUMENT_ROOT . 'config' . DIRECTORY_SEPARATOR;

    if (!is_readable($path . 'config.php')) {
        throw new RuntimeException(
            'Unable to read config from ' . $path . 'config.php'
        );
    }

    $config = include_once $path . 'config.php';

    if (is_array($config)) {
        $config = new Config($config);
    }

    if (!$config instanceof Config) {
        $type = gettype($config);
        if ($type == 'boolean') {
            $type .= ($type ? ' (true)' : ' (false)');
        } elseif (is_object($type)) {
            $type = get_class($type);
        }

        throw new RuntimeException(
            sprintf(
                'Unable to read config file. Config must be either an array or %s instance. Got %s',
                Config::class,
                $type
            )
        );
    }

    if (is_readable($path . APPLICATION_ENV . '.php')) {
        $override = include_once $path . APPLICATION_ENV . '.php';

        if (is_array($override)) {
            $override = new Config($override);
        }

        if ($override instanceof Config) {
            $config->merge($override);
        }
    }

    return $config;
}
```

This allow to use general `config.php` and override it by `develop.php` or `stage.php` :)
```php
$di->setShared('config', $this-> initConfig());
```

**Comment by dschissler** [comment]
Ok while that might be nice it doesn't seem to fit the barebones style of the DevTools templates.  I'll look at that for my project  but hat about my last question specifically regarding the DevTools template?

**Comment by sergeyklay** [comment]
Implemented in `3.0.x`
